src/components/button.tsx

```diff
@@ -44,88 +42,143 @@ const buttonVariants = cva(
       >
-        <div className="pointer-events-auto relative flex cursor-pointer items-center gap-4 md:gap-8">
-          <span
-            className={cn("lowercase transition-all duration-200", {
-              "text-foreground": variantColor === "black",
-              "text-white": variantColor === "white",
-              "group-hover:text-black":
-                ["default", "secondary"].includes(variant ?? "") &&
-                variantColor === "white",
-              "group-hover:text-white":
-                ["default", "secondary"].includes(variant ?? "") &&
-                variantColor === "black",
-              "underline-animation flex": variant === "link",
-            })}
-          >
-            {children}
-          </span>
-
-          {(variant === "gradient" || variant === "secondary") && (
-            <div className="relative grid h-6 w-6 place-items-center md:h-8 md:w-8">
-              <div
-                className={cn(
-                  "h-2 w-2 rounded-full transition-transform duration-75 group-hover:scale-0",
-                  {
-                    "bg-gradient-main": variant === "gradient",
-                    "bg-white":
-                      variant === "secondary" && variantColor === "white",
-                    "bg-black":
-                      variant === "secondary" && variantColor === "black",
-                  },
-                )}
-              />
-              <div className="absolute grid h-full w-full rotate-[30deg] scale-0 place-items-center rounded-full bg-black transition-transform duration-150 ease-out group-hover:animate-reveal-arrow group-hover:ease-in">
-                <ArrowRight className="!size-5 text-white md:!size-6" />
-              </div>
-            </div>
-          )}
-        </div>
+        <ButtonContent
+          variant={variant as Variant}
+          size={size as Size}
+          variantColor={variantColor as VariantColor}
+          disabled={disabled as boolean}
+        >
+          {children}
+        </ButtonContent>
       </Comp>
     );
   },
 );
 Button.displayName = "Button";
 
+function ButtonContent({
+  children,
+  variantColor,
+  variant,
+  disabled,
+}: VariantProps<typeof buttonVariants> & {
+  children: React.ReactNode;
+} & React.ButtonHTMLAttributes<HTMLButtonElement> & {
+    variant?: Variant;
+    variantColor?: VariantColor;
+    size?: Size;
+  }) {
+  return (
+    <div
+      className={cn(
+        "pointer-events-auto relative flex w-full cursor-pointer items-center justify-between gap-4 md:gap-8",
+        {
+          "pointer-events-none": disabled,
+        },
+      )}
+    >
+      <span
+        className={cn("lowercase transition-all duration-200", {
+          "text-foreground": variantColor === ("black" as string),
+          "text-white": variantColor === ("white" as string),
+          "group-hover:text-black":
+            ["default", "secondary"].includes((variant ?? "") as string) &&
```

Comment:
nie rozumiem też trochę po co te wszystkie castowania, bo one nic nie zmieniały z tego co widziałem
```suggestion
            ["default", "secondary"].includes((variant ?? "")) &&
```

Reply:
a wiesz co krzyczał mi eslint i sie budowac nie chciało

Reply:
to dziwne, bo teraz mi nie krzyczał przy tym, ale idk, magia czasami się tu dzieje 😭

Reply:
no wszyscy kochamy ts